nitro-protocol/test/tx-expect-wrappers.ts

```diff
@@ -0,0 +1,32 @@
+/**
+ * Wrapper for transactions that are expected to succeed with no return values.
+ */
+export async function expectSucceed(fn: () => void) {
+  const txResult = (await fn()) as any;
+
+  expect(txResult.length).toBe(0);
+}
+
+/**
+ * Wrapper for calls to `stateIsSupported` that are expected to succeed.
+ */
+export async function expectSupportedState(fn: () => void) {
+  const txResult = (await fn()) as any;
+
+  // `.stateIsSupported` returns a (bool, string) tuple
+  expect(txResult.length).toBe(2);
+  expect(txResult[0]).toBe(true);
+  expect(txResult[1]).toBe('');
+}
+
+/**
+ * Wrapper for calls to `stateIsSupported` that are expected to fail.
+ * Checks that the reason for failure matches the supplied `reason` string.
+ */
+export async function expectFailure(fn: () => void, reason?: string) {
```

Comment:
Should we write a wrapper that will pass a test if the call results in _either_ a revert or a `false` return value? Or do you think it is OK for the tests to "know" which failure mode the implementation has plumped for?

Reply:
I think this is OK as is given current usage. Implementing the other wrapper shouldn't be much burden at time of need.

Did clarify intent here: 6dc6765f